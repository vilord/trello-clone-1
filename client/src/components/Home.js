import React from 'react';
import PropTypes from 'prop-types';
import BoardCollection from '../components/BoardCollection';
import './Home.css';

const Home = ({ boards }) => {
  const hasFavorites = boards.some(x => x.favorite);
  const favoriteBoards = boards.filter(x => x.favorite);
  return (
    <div className="Home">
      {hasFavorites ? (
        <BoardCollection
          icon="empty star"
          title="Starred Boards"
          boards={favoriteBoards}
        />
      ) : null}
      <BoardCollection
        icon="user outline"
        title="Personal Boards"
        boards={boards}
      />
    </div>
  );
};

Home.displayName = 'Home';

Home.propTypes = {
  boards: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
      favorite: PropTypes.bool.isRequired,
      theme: PropTypes.object,
    }),
  ).isRequired,
};

export default Home;
