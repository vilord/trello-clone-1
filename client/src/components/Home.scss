.Home {
  position: fixed;
  padding-top: 10px;
  top: 40px;
  right: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  width: 100%;
  align-items: center;
}
