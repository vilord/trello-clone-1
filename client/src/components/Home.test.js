import React from 'react';
import { shallow } from 'enzyme';

import Home from './Home';

describe('Home Component', () => {
  beforeAll(() => {
    console.error = err => {
      throw new Error(err);
    };
  });

  let minProps;
  let component;
  beforeEach(() => {
    minProps = {
      boards: [
        {
          id: '',
          title: 'Trello Clone',
          favorite: false,
          theme: {
            color: 'blue',
            picture: '',
          },
        },
      ],
    };
    component = shallow(<Home {...minProps} />);
  });

  it('renders', () => {
    expect(component).toHaveLength(1);
  });

  it("doesn't have Starred Boards when there's no favorites", () => {
    expect(
      component
        .find('BoardCollection')
        .someWhere(x => x.prop('title') === 'Starred Boards'),
    ).toBe(false);
  });

  it('has Starred Boards when favorites', () => {
    component.setProps({
      boards: [
        {
          ...minProps.boards[0],
          favorite: true,
        },
      ],
    });
    expect(
      component
        .find('BoardCollection')
        .someWhere(x => x.prop('title') === 'Starred Boards'),
    ).toBe(true);
  });

  it('matches', () => {
    expect(component).toMatchSnapshot();
  });
});
