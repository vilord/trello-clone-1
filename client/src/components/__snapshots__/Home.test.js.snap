// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches 1`] = `
<div
  className="Home"
>
  <BoardCollection
    boards={
      Array [
        Object {
          "favorite": false,
          "id": "",
          "theme": Object {
            "color": "blue",
            "picture": "",
          },
          "title": "Trello Clone",
        },
      ]
    }
    icon="user outline"
    title="Personal Boards"
  />
</div>
`;
